### PR TITLE
Added CommitmentPermissions package

### PR DESCRIPTION
**Pull request by elliotritchie**

### CODE REVIEW

File: src/SFA.DAS.Authorization.NetCoreTestHarness/Models/Party.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+namespace SFA.DAS.Authorization.NetCoreTestHarness.Models
+{
+    public static class Party
+    {
+        public const string Type = "Employer";
+        public const long Id = 1;
```
Comment (Pauljgraham) [inline src/SFA.DAS.Authorization.NetCoreTestHarness/Models/Party.cs]: Should PartyId be a string

File: src/SFA.DAS.Authorization.CommitmentPermissions/AuthorizationContextExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            authorizationContext.Set(AuthorizationContextKey.CohortId, cohortId);
+            authorizationContext.Set(AuthorizationContextKey.PartyType, partyType);
+            authorizationContext.Set(AuthorizationContextKey.PartyId, partyId);
+        }
+        
+        internal static (long CohortId, PartyType PartyType, long PartyId) GetCommitmentPermissionValues(this IAuthorizationContext authorizationContext)
+        {
+            var cohortId = authorizationContext.Get<long?>(AuthorizationContextKey.CohortId);
+            var partyType = authorizationContext.Get<PartyType?>(AuthorizationContextKey.PartyType);
+            var partyId = authorizationContext.Get<long?>(AuthorizationContextKey.PartyId);
```
Comment (Pauljgraham) [inline src/SFA.DAS.Authorization.CommitmentPermissions/AuthorizationContextExtensions.cs]: should this be a string

File: src/SFA.DAS.Authorization.NetCoreTestHarness/Models/Party.cs (hunk 1 above)
Comment (elliotritchie) [reply src/SFA.DAS.Authorization.NetCoreTestHarness/Models/Party.cs]: Fixed

File: src/SFA.DAS.Authorization.CommitmentPermissions/AuthorizationContextExtensions.cs (hunk 2 above)
Comment (elliotritchie) [reply src/SFA.DAS.Authorization.CommitmentPermissions/AuthorizationContextExtensions.cs]: Fixed

File: src/SFA.DAS.Authorization.CommitmentPermissions/AuthorizationContextExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            if (cohortId == null)
+            {
+                throw new InvalidOperationException($"Value for authorization context key '{AuthorizationContextKey.CohortId}' cannot be null");
+            }
+            
+            if (partyType == null)
+            {
+                throw new InvalidOperationException($"Value for authorization context key '{AuthorizationContextKey.PartyType}' cannot be null");
+            }
+
```
Comment (Pauljgraham) [inline src/SFA.DAS.Authorization.CommitmentPermissions/AuthorizationContextExtensions.cs]: Can partyType even be null?

Comment (elliotritchie) [reply src/SFA.DAS.Authorization.CommitmentPermissions/AuthorizationContextExtensions.cs]: Fixed

**Pull request merged**